### ISSUE TITLE
Fix token_f1_per_class metric

### ISSUE THREAD

**Issue by nirhutnik** [deepchecks/deepchecks#2450]
In NLP SingleDatasetPerformance, the token_f1_per_class doesn't work.

**Comment by AmanSal1** [comment]
@nirhutnik can you please give a small code to reproduce the error

**Comment by AmanSal1** [comment]
@nirhutnik I tried a small code of  SingleDatasetPerformance but f1_per_class is working fine
ps: I am attaching the output snippet

![1](https://user-images.githubusercontent.com/121633121/233643397-42261404-094e-439b-8d82-bece79dc675a.PNG)

**Comment by nirhutnik** [comment]
Hi @AmanSal1 ! Thanks for looking at this issue!

The issue is with "token_f1_per_class", not "f1_per_class" which does work fine :)

Here's an example code:

```
# %%
import typing as t
from datasets import load_dataset

from deepchecks.nlp.text_data import TextData

# %%
wikiann = t.cast(t.Any, load_dataset("wikiann", "en"))

# %%
def ner_index_to_ner_tag(index):
    return {
        0: "O",
        1: "B-PER", 
        2: "I-PER",
        3: "B-ORG", 
        4: "I-ORG",
        5: "B-LOC", 
        6: "I-LOC"
    }[index]


train_data = wikiann["train"][:20]

train_dataset = TextData(
    tokenized_text=train_data["tokens"],
    label=[
        [ner_index_to_ner_tag(it) for it in sample]
        for sample in train_data["ner_tags"]
    ],
    task_type="token_classification",
)

from deepchecks.nlp.checks import SingleDatasetPerformance

check = SingleDatasetPerformance(scorers=['token_f1_per_class'])
check.run(train_dataset, predictions=train_dataset.label)
```

**Comment by nirhutnik** [comment]
@AmanSal1 However, I see that there's already an open PR for this issue and is about to be fixed - #2458 

If you'd still like to contribute, I'd be happy to try and suggest another issue to contribute. Then, we can assign it to you so no one else takes it in the meantime :)

**Comment by AmanSal1** [comment]
@nirhutnik yes yes you can assign me it would be grateful

**Comment by noamzbr** [comment]
Closed by #2458
